src/machine-c-api.h

```diff
@@ -82,1894 +83,528 @@ typedef enum { // NOLINT(modernize-use-using)
-/// \returns Pointer to new default config object
-/// \details Object acquired from this function must not be changed and
-/// must be deleted with cm_delete_machine_config. To make a custom config based
-/// on default config user must make a deep copy of returned object members and then
-/// customize
-CM_API const cm_machine_config *cm_new_default_machine_config(void);
-
-/// \brief Delete machine config acquired from cm_new_default_machine_config
-/// \returns void
-CM_API void cm_delete_machine_config(const cm_machine_config *config);
-
-/// \brief Create new machine instance from configuration
-/// \param config Machine configuration. Must be pointer to valid object
-/// \param runtime_config Machine runtime configuration. Must be pointer to valid object
-/// \param new_machine Receives the pointer to new machine instance
-/// \param err_msg Receives the error message if function execution fails
-/// or NULL in case of successful function execution. In case of failure error_msg
-/// must be deleted by the function caller using cm_delete_cstring.
-/// err_msg can be NULL, meaning the error message won't be received.
-/// \returns 0 for success, non zero code for error
-CM_API int cm_create_machine(const cm_machine_config *config, const cm_machine_runtime_config *runtime_config,
-    cm_machine **new_machine, char **err_msg);
-
-/// \brief Create machine instance from previously serialized directory
-/// \param dir Directory where previous machine is serialized
-/// \param runtime_config Machine runtime configuration. Must be pointer to valid object
-/// \param new_machine Receives the pointer to new machine instance
-/// \param err_msg Receives the error message if function execution fails
-/// or NULL in case of successful function execution. In case of failure error_msg
-/// must be deleted by the function caller using cm_delete_cstring.
-/// err_msg can be NULL, meaning the error message won't be received.
-/// \returns 0 for success, non zero code for error
-CM_API int cm_load_machine(const char *dir, const cm_machine_runtime_config *runtime_config, cm_machine **new_machine,
-    char **err_msg);
-
-/// \brief Serialize entire state to directory
-/// \param m Pointer to valid machine instance
+// -----------------------------------------------------------------------------
+
+/// \brief Creates a new machine instance from configuration.
+/// \param config Machine configuration as a JSON string.
+/// \param runtime_config Machine runtime configuration as a JSON string.
+/// \param new_machine Receives the pointer to new machine instance.
+/// \returns 0 for success, non zero code for error.
+CM_API int cm_create(const char *config, const char *runtime_config, cm_machine **new_machine);
+
+/// \brief Destroys a machine.
+/// \returns 0 for success, non zero code for error.
+/// \details The machine is deallocated and its pointer cannot be used after this call.
+CM_API int cm_destroy(cm_machine *m);
```

Comment:
We can set the pointer back to NULL depending on how paranoid you want to be.
```
CM_API int cm_destroy(cm_machine **m);
```

Reply:
Meh.

Reply:
Yea, not doing this hehe